### PR TITLE
Get rid of For loop

### PR DESCRIPTION
**Pull request by MeWu-IDM**
@rhanIDM  I wrote the "get_mortality" from results which does not need to loop over each trials, I tried this and I think the results should be the same as your calculation, let me know if you have queations.
I also found mixed use of dplyr and plyr sometimes cause issues so I think it's better specifying namespace if both are used.

### CODE REVIEW

**Comment by MeWu-IDM** [comment]
@celiot-IDM this was what I mentioned in the meeting, it'll be good if we have a function to return mortality/fertility for all trials

File: analysis_scripts/Run_Sim_Plot_Population.R [hunk 1]
```diff
@@ -11,6 +11,38 @@ library(tidyverse)
...
+                                pivot_longer(x, colnames(x)[grepl("X", colnames(x))], 
+                                             names_prefix = "X", 
+                                             names_to = "Year", 
+                                             values_to = "MortalityRate")
+                              }
+  )
+  # Combine all trials and join to the custom AgeBand
+  all_trials_matrix <- do.call(data.table::rbindlist, list(all_trials_matrix, idcol="Trial"))
+  mortalityrates <- all_trials_matrix %>% 
+    inner_join(agebands)
```
Comment (celiot-IDM) [inline analysis_scripts/Run_Sim_Plot_Population.R:42]: Here's a faster way to generate the all_trials_matrix dataframe (albeit with slightly different column and sort orders):

```
trials <- names(results)
l <- lapply(trials, function(trial){
  t <- results[[trial]][["PopulationRates"]][[categoryName]][["ratesMatrix"]]
  t <- reshape2::melt(t, varnames = c("Label", "Year"), value.name = "MortalityRate")
  t$Trial = rep(trial, nrow(t))
  return(t)
})

all_trials_matrix <- data.table::rbindlist(l)
all_trials_matrix$Label <- as.character(all_trials_matrix$Label)
all_trials_matrix$Year <- as.character(all_trials_matrix$Year)
```

**Review by celiot-IDM — CHANGES_REQUESTED** [review]
See suggested change ...

**Comment by celiot-IDM** [comment]
> @celiot-IDM this was what I mentioned in the meeting, it'll be good if we have a function to return mortality/fertility for all trials

@MeWu-IDM : Are you proposing that I turn the fragment I just sent (or something like it) into a built-in pacehrh:: function?

**Comment by MeWu-IDM** [comment]
@celiot-IDM I was thinking something similar like SaveSuiteDemographics for mortality rates (SaveSuiteMortality)
I will discuss with @rhanIDM and see what type of output format is useful for researchers. I haven't looked closely at internal implementation of results, I thought it may save us time if the internal data structure is easier to work with.

**Comment by MeWu-IDM** [comment]
this has been implemented in #210

**Pull request closed without merging**